[HandleError] rename msg field to action

savil opened the pull request:
this was a TODO and also briefly discussed in https://github.com/rust-bitcoin/rust-lightning/pull/43#pullrequestreview-135649551

I'm not fully sure how to remove the `Option`, and make it completely required. Would love suggestions. So, have omitted that for now. Plus, better to make smaller, incremental changes.

Test Plan:
`cargo build`
`cargo test`

ariard commented on src/ln/msgs.rs:
```diff
@@ -380,7 +380,7 @@ pub enum ErrorAction {
 pub struct HandleError { //TODO: rename me
 	pub err: &'static str,
-	pub msg: Option<ErrorAction>, //TODO: Make this required and rename it
+	pub action: Option<ErrorAction>, //TODO: Make this required and rename it
```
Update the TODO maybe :)

TheBlueMatt replied:
Can remove the "rename it" TODO :)

TheBlueMatt commented:
Note that "making it required" would be s/Option<ErrorAction>/ErrorAction/, but that will require filling in each action, which means figuring out exactly how we want to handle everything.

ariard commented:
To remove the Option I think we need to fulfill incrementally all empty msg field of HandleError with actual errors or futures ones (like BanPeer or KeyError). I'll try to clean others TODO errors in `channelmanager` after #43

savil commented:
- cool,  I updated the TODO. Let me know if anything else needs to be done.

- sorry for the commit thrash, I'm still figuring out the workflow for pull-requests on GitHub. I think I messed up git history for this branch, trying to resolve a merge-conflict from upstream. I updated my local `master` from upstream, and then did `git rebase master` to resolve the conflict. Needed to `git push -f origin <my branch>`, but surely there's a better way that avoids using `-f`. 

- to remove `Option`, agree that it'll need to be incremental. I'm so new to this, I don't feel I have enough context on each call site to figure out what kind of Error to insert there.

ariard commented:
Have a look on Chap2,3,6 : http://labs.kernelconcepts.de/downloads/books/Pro%20Git%20-%20Scott%20Chacon.pdf ;)
You can squash your commits with git rebase -i HEAD~n (with n numbers of commits) to clean your git history.

Anyways don't hesitate to show up on IRC #rust-bitcoin, Freenode

TheBlueMatt commented:
No, force-pushing to a feature branch is generally fine, its only really bad practice on a master branch (though I use my own personal fork's master branch as a feature branch, too, just don't force push to rust-bitcoin/rust-lightning).

TheBlueMatt commented on src/ln/msgs.rs:
```diff
@@ -380,7 +380,7 @@ pub enum ErrorAction {
 
 pub struct HandleError { //TODO: rename me
```
Hmm, I think eventually we need to separate out our error types, making this more of a MessageHandleError or PeerHandleError or something like that, and use a different type where that doesn't make sense.
